packages/salesforcedx-vscode-core/src/commands/forceSourcePull.ts

```diff
@@ -48,9 +69,9 @@ const workspaceChecker = new SfdxWorkspaceChecker();
 const parameterGatherer = new EmptyParametersGatherer();
 
 export async function forceSourcePull(this: FlagParameter<string>) {
-  // tslint:disable-next-line:no-invalid-this
-  const flag = this ? this.flag : undefined;
-  const executor = new ForceSourcePullExecutor(flag);
+  const {flag, commandVersion} = this || {};
```

Comment:
nit: I don't think you need the `|| {}` here.

Reply:
Yes this is needed when the handler is invoked without a flag param in which case it is `undefined` and then de-structure errors

Reply:
We discussed in slack.  The weird part of this is that _shouldn't_ be possible but due to how the registerCommand function works we lose type safety for the parameters when this function is executed.